tests/integration/piv_worker_test.py

```diff
@@ -0,0 +1,228 @@
+# coding: utf8
+#
+# Copyright (c) 2020, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# [matrix] channel #navitia:matrix.org (https://app.element.io/#/room/#navitia:matrix.org)
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+from kirin import app, db
+from kirin.command.piv_worker import PivWorker
+from kirin.core.model import RealTimeUpdate
+from kirin.core.types import ConnectorType
+from kirin.piv.piv import get_piv_contributor
+
+from amqp.exceptions import NotFound
+from kombu import Connection, Exchange, Queue
+import pytest
+import threading
+from retrying import Retrying
+
+
+wait_until = Retrying(stop_max_delay=20000, wait_exponential_multiplier=100)
+contributor_id = "realtime.wuhan"
+exchange_name = "piv"
+queue_name = "piv_kirin"
+
+
+def fassert(test):
+    """
+    `assert` is not supported into `lambda` function.
+    `lambda: assert True == False` is a syntax error.
+    With this function, you can replace it with
+    `lambda: fassert(True == False)`
+    """
+    assert test
```

Comment:
Not sure it's as easy: as you always use them together, can't you merge `wait_until` and `fassert` ?

```suggestion
@retry(stop_max_delay=20000, wait_exponential_multiplier=100)
def wait_until(test):
    assert test
```

Reply:
Much better indeed although, I had to tweak it a little bit: `wait_until` function can't take a boolean but must take a function because it needs to be re-evaluated on every retry. Done in 7caa2336073dd4d08a5b354276066f064a81cbbc.

Reply:
Done